viz_scripts/ebike_specific_metrics.ipynb

```diff
@@ -144,93 +110,87 @@
+    "expanded_ct = scaffolding.energy_impact_kWH(expanded_ct, 'distance_miles', 'Replaced_mode', 'Mode_confirm')\n",
+    "expanded_ct = scaffolding.CO2_impact_lb(expanded_ct, 'distance_miles', 'Replaced_mode', 'Mode_confirm')"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "coral-bicycle",
+   "id": "empty-intensity",
    "metadata": {},
    "outputs": [],
    "source": [
-    "dic_pur"
+    "file_suffix = scaffolding.get_file_suffix(year, month, program)\n",
+    "quality_text = scaffolding.get_quality_text(participant_ct_df, expanded_ct)"
    ]
   },
   {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "changed-aquarium",
+   "cell_type": "markdown",
+   "id": "surgical-continuity",
    "metadata": {},
-   "outputs": [],
    "source": [
-    "assert len(expanded_ct[(expanded_ct['Mode_confirm'] == 'Pilot ebike') & (expanded_ct[\"Replaced_mode\"] == \"Pilot ebike\")]) == 0"
+    "## Metrics for Specific Mode"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "every-stanley",
+   "id": "romance-green",
    "metadata": {},
    "outputs": [],
    "source": [
-    "# Energy Impact Calculation\n",
-    "scaffolding.unit_conversions(expanded_ct)"
+    "data_eb = expanded_ct.query(f\"Mode_confirm == '{mode_of_interest}'\")"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "empty-intensity",
+   "id": "seeing-court",
    "metadata": {},
    "outputs": [],
    "source": [
-    "file_suffix = scaffolding.get_file_suffix(year, month, program)"
+    "quality_text_mode = scaffolding.get_quality_text_ebike(expanded_ct, data_eb)"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "demanding-picking",
-   "metadata": {},
-   "source": [
-    "# Ebike Data Only"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "romance-green",
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "data_eb = expanded_ct.query(\"Mode_confirm == 'Pilot ebike'\")"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "seeing-court",
+   "id": "loaded-expert",
    "metadata": {},
-   "outputs": [],
    "source": [
-    "quality_text = scaffolding.get_quality_text_ebike(expanded_ct, data_eb)"
+    "### Trips by purpose for specified mode"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
    "id": "respiratory-breach",
-   "metadata": {},
+   "metadata": {
+    "scrolled": false
+   },
    "outputs": [],
    "source": [
-    "labels_tp = data_eb['Trip_purpose'].value_counts(dropna=True).keys().tolist()\n",
-    "values_tp = data_eb['Trip_purpose'].value_counts(dropna=True).tolist()\n",
-    "plot_title=\"Number of trips for each purpose for eBike only\\n%s\" % quality_text\n",
-    "file_name= 'ntrips_ebike_purpose%s.png' % file_suffix\n",
-    "pie_chart_purpose(plot_title,labels_tp,values_tp,file_name)"
+    "if study_type == 'program':\n",
```

Comment:
as we discussed, I think we should not even run this notebook if it is not a program

Reply:
Added exception that is raised when this notebook is run for a study. My understanding is that nbclient will run only up through that cell https://nbclient.readthedocs.io/en/latest/client.html.